addon/components/x-toggle/component.js

```diff
@@ -1,30 +1,71 @@
 import Ember from 'ember';
 import layout from './template';
 
-const { on, run, computed, observer } = Ember;
+const { run, computed } = Ember;
+const a = Ember.A;
 
-export default Ember.Component.extend({
+const xToggle = Ember.Component.extend({
   layout: layout,
-  tagName: 'span',
-  classNameBindings: ['toggled:x-toggle-container-checked', 'disabled:x-toggle-container-disabled'],
-  classNames: ['x-toggle-container'],
-  disabled: false,
-  value: false,
-  toggled: false,
+  tagName: '',
+
   name: 'default',
+  disabled: false,
+  value: 'off',
+  offLabel: 'Off:off',
+  onLabel: 'On:on',
+
+  init() {
+    this._super(...arguments);
+    run.schedule('afterRender', () => {
+      // if value is not set to a valid state suggest a default to the container
+      const {toggled, _onValue, _offValue} = this.getProperties('toggled', '_onValue', '_offValue');
+
+      if(toggled === undefined) {
+        const response = this.ddau('onToggle', {
```

Comment:
I don't think this is related to the issue stated. This is simply to check whether the initialized state for value is valid and if not to "recommend" changing it. If the component is started with a valid value then it will not fire an event.

If however, the initialised state is invalid (overwhelming this is because the value property has not be initialised yet), someone who is listening on `onToggle` with the `action` handler can decide whether to accept or decline the suggestion.

For listeners using the `mut` helper it will initialize the component into the "false" state if the input state is invalid. This is a very common pattern as it allows for the use of an undefined property to be initialized to a valid state to start the lifecycle. It is typically what people want and if they don't then they should use the `action` helper instead.

We potentially could modify the `ddau()` behaviour for the `mut` helper to only respond positively to this use-case when the initialised state was "undefined" but I haven't found that to be super important ... in a vast majority of cases consumers using `mut` want a simple, low-cost way of getting up and running. And a control like a toggle button seems like an ideal case to allow for this